<a href="https://colab.research.google.com/github/JasonRiggins/JasonRiggins/blob/main/Data-Analysis-SQL/Analyze_International_Debt_Statistics/Analysint_International_debt.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# World Bank International Debt Analysis

**Dataset:** The dataset belongs to the [World Bank](https://www.worldbank.org/en/home) and contains information about how much money countries worldwide owe the bank.

Just like citizens go to the bank if they need money to grow economically, contries also approach the world ban for loans to support development in their countries. In the project, we will explore contries' debts to the World Bank using SQL. We will find the answers to questions like:

1. What is the total amount of money that all countries owe to the World Bank?
2. Which country has the highest debt, how much is that, and the percentage of the worlds debt?
3. What is the mean debt owed by countries for different debt indicators?
4. What is the maximum debt held by the top countries by indicator type?

## Connecting to Google BigQuery
Code to make the connection to the world_bank_intl_debt database where the table internation_debt is located. 

In [ ]:
#@title
from google.colab import auth
auth.authenticate_user()

%load_ext google.colab.data_table

The first thing we need to do is select everything in the table to understand how the data is layed out. Let's limit that output to take a quick look.

In [ ]:
%%bigquery --project eco-plating-232402

SELECT * 
FROM `bigquery-public-data.world_bank_intl_debt.international_debt`
ORDER BY country_name
LIMIT 10


Query is running:   0%|          |

Downloading:   0%|          |

country_name country_code            indicator_name  indicator_code  value  \
0  Afghanistan          AFG  IDA grants (current US$)  DT.DIS.IDAG.CD    NaN   
1  Afghanistan          AFG         GNI (current US$)  NY.GNP.MKTP.CD    NaN   
2  Afghanistan          AFG  IDA grants (current US$)  DT.DIS.IDAG.CD    NaN   
3  Afghanistan          AFG  IDA grants (current US$)  DT.DIS.IDAG.CD    NaN   
4  Afghanistan          AFG         GNI (current US$)  NY.GNP.MKTP.CD    NaN   
5  Afghanistan          AFG         GNI (current US$)  NY.GNP.MKTP.CD    NaN   
6  Afghanistan          AFG         GNI (current US$)  NY.GNP.MKTP.CD    NaN   
7  Afghanistan          AFG  IDA grants (current US$)  DT.DIS.IDAG.CD    NaN   
8  Afghanistan          AFG         GNI (current US$)  NY.GNP.MKTP.CD    NaN   
9  Afghanistan          AFG  IDA grants (current US$)  DT.DIS.IDAG.CD    NaN   

   year  
0  <NA>  
1  <NA>  
2  <NA>  
3  <NA>  
4  <NA>  
5  <NA>  
6  <NA>  
7  <NA>  
8  <NA>  
9  <NA>

In [ ]:
%%bigquery --project eco-plating-232402

SELECT COUNT(*) 
FROM `bigquery-public-data.world_bank_intl_debt.international_debt`

Query is running:   0%|          |

Downloading:   0%|          |

f0_
0  3284415

## 1. What is the total amount of money owed to the world bank?
After looking at the results we can see that Afghanistan would have debt amounts with a different indicator_name. But, we don't know how many different countries or indicator names we are working with. Let's find out how many countries there are, the total of the worlds debt.

In [ ]:
%%bigquery --project eco-plating-232402
SELECT 
    COUNT(DISTINCT country_name) AS total_distinct_countries,
    CAST(SUM(value) AS int) AS total_debt,
FROM
  `bigquery-public-data.world_bank_intl_debt.international_debt`

Query is running:   0%|          |

Downloading:   0%|          |

total_distinct_countries       total_debt
0                       135  316944895604019

As we can see form the query above there are 135 countries and a total debt of 316,944,895,604,020 USD on the table. That's a total world debt of 317 trillion dollars.

## Finding the distinct debt indicators

Lets take a look at the debt indicators and codes to see how many there are.

In [ ]:
%%bigquery --project eco-plating-232402
SELECT 
    COUNT(DISTINCT indicator_name) AS number_of_indicator_names, 
    COUNT(DISTINCT indicator_code) AS number_of_indicator_codes
FROM
  `bigquery-public-data.world_bank_intl_debt.international_debt`

Query is running:   0%|          |

Downloading:   0%|          |

number_of_indicator_names  number_of_indicator_codes
0                        497                        497

It seems safe to assume that the indicator codes are directly related to the indicator names. We will be using the indicator_names columnin the rest of the analysis to make things easier to read.

## 2. Which country has the highest debt, how much is that, and the percentage of the worlds debt?


Now that we have an idea of what numbers we are working with let's see which countries are carrying the most debt and the average amounts of the top debt indicator.

<div class="alert alert-block alert-info">
The - country_name NOT LIKE '%income' - line of code was used to eleminate some generic country names that included multiple small countries. This elimated data associated with names like: Low & middle income, Middle income, Upper middle income, and Lower middle income
</div>


In [ ]:
%%bigquery --project eco-plating-232402
SELECT
    country_name,
    CAST(SUM(value) AS int) AS total_debt,
    ROUND(CAST(SUM(value) AS int) / 316944895604019 * 100,2) AS percent_of_world_debt
FROM
    `eco-plating-232402.world_bank_intl_debt.international_ddebt`
    WHERE
        country_name NOT LIKE '%income'
    GROUP BY country_name
    ORDER BY total_debt DESC
LIMIT 10

Query is running:   0%|          |

Downloading:   0%|          |

country_name      total_debt  \
0  Latin America & Caribbean (excluding high income)  21586527231259   
1      Europe & Central Asia (excluding high income)  12470638299106   
2        East Asia & Pacific (excluding high income)   9526097308235   
3                                         South Asia   7518452126433   
4         Sub-Saharan Africa (excluding high income)   7089309388594   
5                                             Mexico   6261206147616   
6                                           IDA only   5934219325574   
7       Least developed countries: UN classification   5489405066449   
8                                              India   4476825177261   
9                                             Brazil   3859450893962   

   percent_of_world_debt  
0                   6.81  
1                   3.93  
2                   3.01  
3                   2.37  
4                   2.24  
5                   1.98  
6                   1.87  
7                   1.73  
8                   1.41  
9                   1.22

The country Latin America & Caribbean is carrying the most debt of any country at 21,586,527,231,259 USD. That is 6.81 percent of the worlds debt and almost double the amout of debt of the next highest country.

## 3. What is the average debt owed for different debt indicators?


In [ ]:
%%bigquery --project eco-plating-232402
SELECT
    indicator_name AS debt_type,
    CAST(AVG(value) AS int) AS indicator_average,   
    ROUND(CAST(SUM(value) AS int) / 316944895604019 * 100,2) AS percent_of_world_debt
FROM
    `eco-plating-232402.world_bank_intl_debt.international_ddebt`
    WHERE
        country_name NOT LIKE '%income'
    GROUP BY debt_type
    ORDER BY indicator_average DESC
LIMIT 10

Query is running:   0%|          |

Downloading:   0%|          |

debt_type  indicator_average  \
0  Principal repayments on external debt, long-te...         4409173910   
1  Principal repayments on external debt, public ...         2841935625   
2  Principal repayments on external debt, private...         1567238285   
3         PPG, official creditors (AMT, current US$)         1474301478   
4          PPG, private creditors (AMT, current US$)         1367634147   
5  Interest payments on external debt, long-term ...         1308319556   
6                      PPG, bonds (AMT, current US$)         1131161658   
7               PPG, multilateral (AMT, current US$)          852352833   
8          PPG, private creditors (INT, current US$)          648632200   
9                  PPG, bilateral (AMT, current US$)          621948646   

   percent_of_world_debt  
0                  10.13  
1                   6.53  
2                   3.60  
3                   3.39  
4                   3.14  
5                   3.01  
6                   2.60  
7                   1.96  
8                   1.49  
9                   1.43


The highest indicator is Principal repayments on external debt, long-term at an overall average of 4,409,173,910 USD. This indicator accounts for 10.13 percent of the world's total debt.


## 4. What is the maximum debt by indicator type in respect to country names?

In [ ]:
%%bigquery --project eco-plating-232402
SELECT
    country_name,
    indicator_name,
    CAST(Max(value) AS int) AS maximum_debt
FROM
    `eco-plating-232402.world_bank_intl_debt.international_ddebt`
WHERE
    country_name NOT LIKE '%income'
GROUP BY country_name, indicator_name
ORDER BY maximum_debt DESC
LIMIT 10

Query is running:   0%|          |

Downloading:   0%|          |

country_name  \
0  Latin America & Caribbean (excluding high income)   
1      Europe & Central Asia (excluding high income)   
2  Latin America & Caribbean (excluding high income)   
3  Latin America & Caribbean (excluding high income)   
4        East Asia & Pacific (excluding high income)   
5  Latin America & Caribbean (excluding high income)   
6  Latin America & Caribbean (excluding high income)   
7                                         South Asia   
8      Europe & Central Asia (excluding high income)   
9      Europe & Central Asia (excluding high income)   

                                      indicator_name  maximum_debt  
0  Principal repayments on external debt, long-te...   82521070084  
1  Principal repayments on external debt, long-te...   63558133854  
2  Principal repayments on external debt, public ...   59445236262  
3          PPG, private creditors (AMT, current US$)   43513895536  
4  Principal repayments on external debt, long-te...   43480460753  
5                      PPG, bonds (AMT, current US$)   40118171134  
6  Interest payments on external debt, long-term ...   36214576148  
7  Principal repayments on external debt, long-te...   35177855642  
8  Principal repayments on external debt, private...   31905991216  
9  Principal repayments on external debt, public ...   31652142638

According to the query Latin America & Caribbean and Europe & Central Asia with the debt type of Principal repayments on external debt, long-term are within 21 billion USD of each other.